### PR TITLE
initial sample data

### PR DESCRIPTION
**Pull request by kevinyamauchi**
this PR establishes the config and class structure for the sample data.

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
Example config:

```python
dataset_description = (
    "3D reconstruction of HIV virus-like particles from cryo-electron"
    "tomography data in EMPIAR-10164 and associated particle poses."
)

hiv_config = {
    "name": "hiv",
    "author": "Alister Burt",
    "description": dataset_description,
    "base_url": "doi:10.5281/zenodo.6504891/",
    "tomogram": {
        "file_name": "01_10.00Apx.mrc",
        "checksum": "md5:426325d006fe04276ea01df9d83ad510",
        "reader": mrc_reader,
    },
}
```


Here's the current repr/str:

```
cryoET DataSet
    name: hiv
    author: Alister Burt
    base url: doi:10.5281/zenodo.6504891/

    Description:
        3D reconstruction of HIV virus-like particles from cryo-
        electrontomography data in EMPIAR-10164 and associated particle poses.

    data
      └── tomogram
          ├── file name: 01_10.00Apx.mrc
          └── checksum: md5:426325d006fe04276ea01df9d83ad510
```

**Comment by kevinyamauchi** [comment]
@alisterburt , do you think we should consolidate `DataSet` and the `DataSetConfig` (pydantic model) into a single pydantic model? The upside of this is removing the `DataSet._config`. The downside, is that we lose the separation between the data set definition (i.e., the config) and the implementation (i.e., the pooch downloader). I could imagine wanting a different `DataSet` implementation that streams a zarr file into a lazy (e.g., dask) array. That being said, it feels like we won't be moving away from pooch any time soon.

I suppose another option is to have a `PoochDataSet` that is a subclass of `BaseDataSet` (which is the config).

I think that we don't need to overoptimize here - I think all of the above options work and I think we would be happy with any of them. I'm leaning towards combining the `DataSet` and `DataSetConfig` into a single model, which I think is what you had originally suggested. 

Thoughts?

**Comment by kevinyamauchi** [comment]
@alisterburt , I've integrated config and downloader into the same class. As mentioned above, I think this is cleaner and I think we can factor out the config in a non-breaking way pretty easily.

This is ready for review! I think we can focus on the data model in this PR and then make the napari plugin in a follow up. The only thing that would be nice to add in the future is a downloader test. Maybe we can upload a tiny fake test mrc file to zenodo or this repo for testing in the future (#2 ).

File: src/cryo_et_sample_data/_tests/test_data_set.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import pytest
+from pydantic import ValidationError
+
+from cryo_et_sample_data._data_set import DataSet, FileMetadata
+
+
+def test_file_metadata():
+    metadata = FileMetadata(
+        file_name="test.zarr",
+        checksum="mic_check_one_two",
```
Comment (alisterburt) [inline src/cryo_et_sample_data/_tests/test_data_set.py:11]: 😆

File: src/cryo_et_sample_data/_tests/test_data_set.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        metadata.reader = mrcfile.open
+
+
+def test_invalid_file_metadata_reader():
+    """FileMetadata.reader must be a callable"""
+    with pytest.raises(ValidationError):
+        _ = FileMetadata(
+            file_name="test.zarr",
+            checksum="mic_check_one_two",
+            reader="mrcfile.read",
```
Comment (alisterburt) [inline src/cryo_et_sample_data/_tests/test_data_set.py]:
```suggestion
            reader="not a callable",
```

Comment (alisterburt) [reply src/cryo_et_sample_data/_tests/test_data_set.py]: just for readability

File: src/cryo_et_sample_data/_hiv.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import mrcfile
```
Comment (alisterburt) [inline src/cryo_et_sample_data/_hiv.py:6]:
```suggestion
"""Immature HIV-1 virus-like particle data from EMPIAR-10164.

This dataset contains a tomogram reconstructed from data available in EMPIAR-10164.
"""

import mrcfile
```

File: src/cryo_et_sample_data/_hiv.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+dataset_description = (
+    "3D reconstruction of HIV virus-like particles from cryo-electron"
+    " tomography data in EMPIAR-10164 and associated particle poses."
+)
+
+hiv_config = {
+    "name": "hiv",
+    "author": "Alister Burt",
+    "description": dataset_description,
```
Comment (alisterburt) [inline src/cryo_et_sample_data/_hiv.py]:
```suggestion
    "description": __doc__,
```

File: src/cryo_et_sample_data/_hiv.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import mrcfile
+
+from cryo_et_sample_data._data_set import DataSet
+
+dataset_description = (
+    "3D reconstruction of HIV virus-like particles from cryo-electron"
+    " tomography data in EMPIAR-10164 and associated particle poses."
+)
+
```
Comment (alisterburt) [inline src/cryo_et_sample_data/_hiv.py]:
```suggestion
```

**Review by alisterburt — APPROVED** [review]
This is looking fantastic @kevinyamauchi - I immediately know how to extend it which is ideal! Thanks for the effort here

Happy to merge as is but have made the super minor suggestion of describing the dataset in the module docstring and referencing that when creating the config

re your question about config vs integrated - I'm easy but quite like how it feels as is!

**Pull request merged**